numpy/lib/shape_base.py

```diff
@@ -859,14 +859,17 @@ def tile(A, reps):
     else:
         c = _nx.array(A, copy=False, subok=True, ndmin=d)
     shape = list(c.shape)
-    n = max(c.size, 1)
+    n = c.size
     if (d < c.ndim):
         tup = (1,)*(c.ndim-d) + tup
-    for i, nrep in enumerate(tup):
-        if nrep != 1:
-            c = c.reshape(-1, n).repeat(nrep, 0)
-        dim_in = shape[i]
-        dim_out = dim_in*nrep
-        shape[i] = dim_out
-        n //= max(dim_in, 1)
+    if n == 0:
+        shape = tuple(s*t for s, t in zip(shape, tup))
+    else:
+        for i, nrep in enumerate(tup):
+            if nrep != 1:
+                c = c.reshape(-1, n).repeat(nrep, 0)
+            dim_in = shape[i]
```

Comment:
This line and the two following are doing the exact some thing that the `n == 0` branch. The code may become simpler if we run this loop without these three lines `if n > 0`, then doing the multiplication of `shape` and `tup` unconditionally for all inputs.